cms/plugin_rendering.py

```diff
@@ -18,6 +18,8 @@
 from cms.utils.django_load import iterload_objects
 from cms.utils.placeholder import get_toolbar_plugin_struct, restore_sekizai_context
 
+from cms.utils.permissions import has_plugin_permission
+from functools import partial
```

Comment:
as above please add this import under ``from collections import deque`` to separate top-level packages.